### PR TITLE
enhance: Enhance and correct exception module

### PR DESCRIPTION
**Pull request by zhagnlu**
#33704

### CODE REVIEW

**Review by longjiquan — COMMENTED** [review]
/lgtm
Btw, I think we should add the stack trace when we are going to raise an exception. It can be done with the little change of error message in next pr.

File: internal/core/src/common/Types.h [hunk 1]
```diff
@@ -132,8 +132,8 @@ GetDataTypeSize(DataType data_type, int dim = 1) {
         // method must handle this case themselves and must not pass
         // VECTOR_SPARSE_FLOAT data_type.
         default: {
-            throw SegcoreError(DataTypeInvalid,
-                               fmt::format("invalid type is {}", data_type));
+            PanicInfo(DataTypeInvalid,
+                      fmt::format("invalid type is {}", data_type));
```
[the remark this answers is not in the document]
Comment (zhagnlu) [reply internal/core/src/common/Types.h:136]: ok

**Comment by zhagnlu** [comment]
rerun ut

**Comment by zhagnlu** [comment]
rerun ut

**Comment by zhagnlu** [comment]
rerun ut

**Comment by zhagnlu** [comment]
rerun ut

**Pull request merged**